Add docs for Translators on how translations work (#926)
NationalCentreTruthReconciliation/Secure-Record-Transfer: TRANSLATORS.md

  Translations are stored in `locale/<language_code>/LC_MESSAGES/django.po`.

- **JavaScript files:**
  Translations go into `locale/<language_code>/LC_MESSAGES/djangojs.po`.

  Example:
  ```
  locale/
    hi/
      LC_MESSAGES/
        django.po       # Python/template strings
        djangojs.po     # JavaScript strings
  ```

---

## Extracting Strings

To extract translatable strings after making changes:

```bash
# For Python and templates
python manage.py makemessages -l hi

# For JavaScript
python manage.py makemessages -d djangojs -l hi

# Extract for all configured languages at once
python manage.py makemessages -a

# Extract JavaScript strings for all languages
python manage.py makemessages -d djangojs -a
```

Replace `hi` with the target language code (e.g., `fr`, `es`, etc.).

**Useful flags:**
- `-a, --all`: Extract strings for all configured languages
- `-d, --domain`: Specify domain (default: `django`, use `djangojs` for JavaScript)
- `-l, --locale`: Specify a particular language code
- `--help`: See all available options and flags

For a complete list of options, run:
```bash
python manage.py makemessages --help
```

---

## Compiling Translations

> After editing `.po` files, compile them so Django can use the updated translations:
> 
> ```bash
> python manage.py compilemessages
> ```
> 

Reviewer: People probably don't need to run this manually, but I like having this here just in case they do.

```suggestion
After editing `.po` files, compile them so Django can use the updated translations:

```bash
python manage.py compilemessages
```

This is done automatically for you in development when you run the application, but you can also run it manually like this if needed.
```

---

## Python Translations (`gettext_lazy`)

In Python files, we use `gettext_lazy` (commonly aliased as `_`) to mark strings for translation.

We do this because we don’t want these strings to be translated immediately when the code is loaded. Instead, we want the translation to happen **later, when the page is being rendered** — for example, when displaying a model field label or a form error message.

This is important because the user’s language preference might not be known at import time (i.e., when the Python code is first loaded), but will be available at render time. Using `gettext_lazy` ensures that the correct translation is applied based on the active language at that moment.

## JavaScript Translations (`djangojs.po`)

We use `JavaScriptCatalog` to make translations available to frontend code.

You don’t need to worry about the setup — just translate the `djangojs.po` file as you would with `django.po`.

### Behind the scenes:
- `JavaScriptCatalog.as_view()` is set up in `urls.py`
- `{% javascript_catalog %}` is included in templates to load translations

> - Strings in JS files are marked using `gettext()` or `_()`

Reviewer: ```suggestion
- Strings in JS files are marked as translatable using `window.django.gettext()`
```

---

## Complete Translation Workflow

Here's the step-by-step process for translating:

> 1. **Extract new strings**: `python manage.py makemessages -l hi`
> 2. **Edit the `.po` file** with your translations (using Poedit or text editor)
> 3. **Compile translations**: `python manage.py compilemessages`
> 4. **Test your changes** by switching language in the app
> 5. **Submit your translation updates** via pull request

Reviewer: In case someone's not super familiar with GitHub, we can make it clearer that they have to fork the repo first.

```suggestion
1. **Fork this repository** to your own GitHub account
2. **Make a new branch** for your translations
3. **Extract new strings**: `python manage.py makemessages -l hi`
4. **Edit the `.po` file** with your translations (using Poedit or text editor)
5. **Compile translations**: `python manage.py compilemessages`
6. **Test your changes** by switching language in the app
7. **Submit your translation updates** via pull request
```

Author: This makes more sense as to how translators would work with translating the website!

---

## Testing Your Translations

After compiling your translations, you'll want to verify they work correctly:

### Switching Languages
- Look for a language selector in the app interface located in the header of the website

### Verification Checklist
- [ ] Text appears in your target language
- [ ] Placeholders are filled with actual values
- [ ] Text fits properly in the UI (not cut off)
- [ ] Special characters display correctly

### Troubleshooting
If translations don't appear:

> 1. Check you ran `compilemessages` after editing `.po` files
> 2. Restart the development server: `docker compose -f compose.dev.yml restart app`
> 3. If restart doesn't work, rebuild the container: `docker compose -f compose.dev.yml up -d --build app`
> 4. Clear your browser cache
> 5. Verify the `.mo` files were created in the locale directory

Reviewer: The `gettext_lazy` thing might be an issue, it's worth mentioning here:

```suggestion
1. Check you ran `compilemessages` after editing `.po` files
2. Restart the development server: `docker compose -f compose.dev.yml restart app`
3. If restart doesn't work, rebuild the container: `docker compose -f compose.dev.yml up -d --build app`
4. Clear your browser cache
5. Verify the `.mo` files were created in the locale directory
6. Try changing the code to use `gettext_lazy` instead of `gettext`
```
